src/teleop_twist_joy.cpp

```diff
@@ -64,104 +64,121 @@ struct TeleopTwistJoy::Impl
     std::bind(&TeleopTwistJoy::Impl::joyCallback, this->pimpl_, std::placeholders::_1),
     rmw_qos_profile_sensor_data);
 
-  pimpl_->enable_button = 5;
-  node->get_parameter("enable_button", pimpl_->enable_button);
-
-  pimpl_->enable_turbo_button = -1;
-  node->get_parameter("enable_turbo_button", pimpl_->enable_turbo_button);
-
-  // TODO(clalancette): node->get_parameter(s) doesn't seem to
-  // support getting a map of values yet.  Revisit this once it does.
-  // if (nh_param->getParam("axis_linear", pimpl_->axis_linear_map))
-  // {
-  //   nh_param->getParam("axis_linear", pimpl_->axis_linear_map);
-  //   nh_param->getParam("scale_linear", pimpl_->scale_linear_map);
-  //   nh_param->getParam("scale_linear_turbo", pimpl_->scale_linear_turbo_map);
-  // }
-  // else
-  {
-    pimpl_->axis_linear_map["x"] = 5;
-    node->get_parameter("axis_linear", pimpl_->axis_linear_map["x"]);
-    pimpl_->scale_linear_map["x"] = 0.5;
-    node->get_parameter("scale_linear", pimpl_->scale_linear_map["x"]);
-    pimpl_->scale_linear_turbo_map["x"] = 1.0;
-    node->get_parameter("scale_linear_turbo", pimpl_->scale_linear_turbo_map["x"]);
-  }
-
-  // TODO(clalancette): node->get_parameter(s) doesn't seem to
-  // support getting a map of values yet.  Revisit this once it does.
-  // if (nh_param->getParam("axis_angular", pimpl_->axis_angular_map))
-  // {
-  //   nh_param->getParam("axis_angular", pimpl_->axis_angular_map);
-  //   nh_param->getParam("scale_angular", pimpl_->scale_angular_map);
-  //   nh_param->getParam("scale_angular_turbo", pimpl_->scale_angular_turbo_map);
-  // }
-  // else
-  {
-    pimpl_->axis_angular_map["yaw"] = 2;
-    node->get_parameter("axis_angular", pimpl_->axis_angular_map["yaw"]);
-    pimpl_->scale_angular_map["yaw"] = 0.5;
-    node->get_parameter("scale_angular", pimpl_->scale_angular_map["yaw"]);
-    pimpl_->scale_angular_turbo_map["yaw"] = pimpl_->scale_angular_map["yaw"];
-    node->get_parameter("scale_angular_turbo", pimpl_->scale_angular_turbo_map["yaw"]);
+  // TODO(clalancette): node->get_parameter(s) doesn't support getting a map of
+  // values yet.  https://github.com/ros2/rclcpp/issues/529
+  int_parameters.push_back({"enable_button", 5, pimpl_->enable_button});
+  int_parameters.push_back({"enable_turbo_button", -1, pimpl_->enable_turbo_button});
+  int_parameters.push_back({"axis_linear_x", 5, pimpl_->axis_linear_map["x"]});
+  int_parameters.push_back({"axis_angular_yaw", 2, pimpl_->axis_angular_map["yaw"]});
+
+  double_parameters.push_back({"scale_linear_x", 0.5, pimpl_->scale_linear_map["x"]});
+  double_parameters.push_back({"scale_linear_turbo_x", 1.0, pimpl_->scale_linear_turbo_map["x"]});
+  double_parameters.push_back({"scale_angular_yaw", 0.5, pimpl_->scale_angular_map["yaw"]});
+  double_parameters.push_back({"scale_angular_turbo_yaw", 0.5, pimpl_->scale_angular_turbo_map["yaw"]});
+
+  for(const auto & p : int_parameters) {
+    this->set_parameter_if_not_set(p.name, p.default_value);
```

Comment:
True, this is changing the semantic, but I don't think the existing semantic is a good one.  If you start up the node with the existing semantic and no parameters, then 0 parameters will be exposed (that is, `ros2 param list` will return an empty list for this node).  The only way to know that there are parameters that could be set is to read the code, and that just doesn't seem to make much sense.  I'd much rather have the parameters available at all times (and set to their default values at startup), at which point it is discoverable.

> Keeping that semantic seems like a good idea since otherwise you can't distinguish if the parameter was original set to a value or if it only got set from the default in code.

It seems to me that what mostly matters is what the current value for each parameter is (it is not like this is a complex node, after all).  Is there a particular reason that you see where distinguishing this matters?